### PR TITLE
Handle buffers with no buffer-file-name

### PR DESCRIPTION
**Pull request by yyoncho**
- This was reported in gitter by @dsyzling

### CODE REVIEW

File: lsp.el [hunk 1]
```diff
@@ -3290,7 +3292,7 @@ current language. When IGNORE-MULTI-FOLDER is nil current file
 will be openned in multi folder language server if there is
 such."
   (-if-let* ((session (lsp-session))
-             (project-root (lsp--calculate-root session (buffer-file-name)))
+             (project-root (lsp--calculate-root session (or (buffer-file-name) default-directory)))
```
Comment (MaskRay) [inline lsp.el]: Why is `default-directory` used here?

Comment (yyoncho) [reply lsp.el]: When you do find-file and create a new file it won't have buffer-file-name but effectively will be part of the current project.

File: lsp.el [hunk 2]
```diff
@@ -3308,17 +3309,18 @@ Returns nil if the project should not be added to the current SESSION."
...
+    (when-let (workspace (->> (lsp-session)
+                              (lsp--session-workspaces)
+                              (--first
+                               (when-let (library-folders-fn
+                                          (-> it lsp--workspace-client lsp--client-library-folders-fn))
+                                 (-first (lambda (library-folder)
+                                           (f-ancestor-of? library-folder (buffer-file-name)))
+                                         (funcall library-folders-fn it))))))
+      (lsp--open-in-workspace workspace)
+      (view-mode t)
```
Comment (MaskRay) [inline lsp.el]: `(view-mode 1)`

File: lsp.el [hunk 3]
```diff
@@ -637,7 +637,8 @@ This is equivalent to `display-warning', using `lsp-mode' as the type and
   "Return URI of the current buffer."
   (or lsp-buffer-uri
       (lsp--path-to-uri
-       (or buffer-file-name (ignore-errors (buffer-file-name (buffer-base-buffer)))))))
+       (or buffer-file-name (ignore-errors (buffer-file-name (buffer-base-buffer)))
+           (f-join default-directory (buffer-name))))))
```
Comment (MaskRay) [inline lsp.el]: `(expand-file-name (buffer-name))`

**Pull request closed without merging**